kubernetes/config-maps/nginx-proxy/nginx.conf.j2

```diff
@@ -29,3 +29,16 @@ http {
 
     include /etc/nginx/conf.d/*http.conf;
 }
+
+
+# TCP connection
+stream {
+
+    log_format basic '$remote_addr [$time_local] '
+                     '$protocol $status $bytes_sent $bytes_received '
+                     '$session_time';
+
+    access_log /var/log/nginx/access.log basic buffer=32k;
+
+    include /etc/nginx/conf.d/*tcp.conf;
```

Comment:
Please create a separate directory TCP configs. E.g. `tcp.d` and rename `conf.d` to `http.d`. We want the file structure to be obvious and easy to understand.

Reply:
This structure implemented in considered way. We need only two kubernetes configmaps. In your solution they need three.

Reply:
So let's just remove `conf.d` and put `http.conf` and  `tcp.conf` directly in `/etc/nginx`. This way we'll be able to have just one config map for everything.

Reply:
This is also not possible to do. This would work if the mount directory is empty or files within are not needed. When we want mount configmap in `/etc/nginx/` without delete other files, we need three configmaps.

Reply:
In that case let's create a subdirectory and put `http.conf` and `tcp.conf` in it. Just don't call it `conf.d` (or anything ending with `.d`) because that creates an expectation that everything you drop in that directory is simply included in the config.

Reply:
Done.